pkg/git/GitCliUtil.go

```diff
@@ -0,0 +1,83 @@
+package git
+
+import (
+	"fmt"
+	"go.uber.org/zap"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/config"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+type GitUtil struct {
+	logger *zap.SugaredLogger
+}
+
+func NewGitUtil(logger *zap.SugaredLogger) *GitUtil {
+	return &GitUtil{
+		logger: logger,
+	}
+}
+
+const GIT_AKS_PASS = "/git-ask-pass.sh"
+
+func (impl *GitUtil) Fetch(rootDir string, username string, password string) (response, errMsg string, err error) {
+	impl.logger.Debugw("git fetch ", "location", rootDir)
+	cmd := exec.Command("git", "-C", rootDir, "fetch", "origin", "--tags", "--force")
+	output, errMsg, err := impl.runCommandWithCred(cmd, username, password)
+	impl.logger.Debugw("fetch output", "root", rootDir, "opt", output, "errMsg", errMsg, "error", err)
+	return output, "", nil
+}
+
+func (impl *GitUtil) Checkout(rootDir string, branch string) (response, errMsg string, err error) {
+	impl.logger.Debugw("git fetch ", "location", rootDir)
+	cmd := exec.Command("git", "-C", rootDir, "checkout", branch, "--force")
+	output, errMsg, err := impl.runCommand(cmd)
+	impl.logger.Debugw("fetch output", "root", rootDir, "opt", output, "errMsg", errMsg, "error", err)
+	return output, "", nil
+}
+
+func (impl *GitUtil) runCommandWithCred(cmd *exec.Cmd, userName, password string) (response, errMsg string, err error) {
+	cmd.Env = append(os.Environ(),
```

Comment:
it will only be used for AKS? also will be ASKPASS or AKSPASS ?

Reply:
yas for ASK. variable is `ASKPASS`